agent/credentials/manager.go

```diff
@@ -57,6 +57,15 @@ type IAMRoleCredentials struct {
 type TaskIAMRoleCredentials struct {
 	ARN                string
 	IAMRoleCredentials IAMRoleCredentials
+	lock               sync.RWMutex
+}
+
+// GetIAMRoleCredentials returns the iamrole credentials in the task iam role struct
```

Comment:
GetIAMRoleCredentials returns the IAM role credentials in the task IAM role struct.